### PR TITLE
add gatk4/fastqtosam #198

### PR DESCRIPTION
**Pull request by ntoda03**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: software/gatk4/fastqtosam/main.nf [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        gatk FastqToSam \\
+            -F1 ${reads[0]} \\
+            -F2 ${reads[1]} \\
+            -O ${prefix}.bam \\
+            -SM $prefix \\
+            $options.args
+
+        gatk --version | grep Picard | sed "s/Picard Version: //g" > ${software}.version.txt
+        """
+    }
```
Comment (drpatelh) [inline software/gatk4/fastqtosam/main.nf:52]:
```suggestion
    def software   = getSoftwareName(task.process)
    def prefix     = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
    def read_files = meta.single_end ? "-F1 $reads" : "-F1 ${reads[0]} -F2 ${reads[1]}"
    """
    gatk FastqToSam \\
        $read_files \\
        -O ${prefix}.bam \\
        -SM $prefix \\
        $options.args
    gatk --version | grep Picard | sed "s/Picard Version: //g" > ${software}.version.txt
    """
```

Comment (drpatelh) [reply software/gatk4/fastqtosam/main.nf:52]: How about this instead @ntoda03 ?

**Review by drpatelh — CHANGES_REQUESTED** [review]
Looks amazing @ntoda03 ! One minor change for simplicity.

File: software/gatk4/fastqtosam/main.nf (hunk 1 above)
Comment (ntoda03) [reply software/gatk4/fastqtosam/main.nf]: @drpatelh Ah yes that's a smart way to do it

**Review by drpatelh — APPROVED** [review]

**Pull request merged**